mne/annotations.py

```diff
@@ -282,8 +283,35 @@ def __init__(
         self.onset, self.duration, self.description, self.ch_names = _check_o_d_s_c(
             onset, duration, description, ch_names
         )
+        self._id = np.array(
+            [str(uuid.uuid4()) for _ in range(len(self.onset))], dtype=str
+        )
         self._sort()  # ensure we're sorted
 
+    # We need this, bc if we create Annotations from events which are not sorted,
+    # and we then directly query for the according IDs, we cannot do this simply by
+    # index as the Annotations will already on creation be sorted.
+    def _get_id(self, onset, duration, description):
```

Comment:
I'm afraid that this is far from optimal and might be slow once there are many annotations.